editoast/src/infra_cache/mod.rs

```diff
@@ -474,27 +476,35 @@ impl InfraCache {
     }
 
     /// Apply delete operation to the infra cache
-    pub fn apply_delete(&mut self, object_ref: &ObjectRef) {
+    pub fn apply_delete(&mut self, object_ref: &ObjectRef) -> Result<()> {
         let obj_cache = self.objects[object_ref.obj_type]
             .remove(&object_ref.obj_id)
-            .unwrap();
+            .ok_or_else(|| OperationResultError::ObjectNotFound {
+                obj_type: object_ref.obj_type.to_string(),
+                obj_id: object_ref.obj_id.clone(),
+            })?;
 
         for track_id in obj_cache.get_track_referenced_id() {
             self.track_sections_refs
                 .get_mut(track_id)
-                .unwrap()
+                .ok_or_else(|| OperationResultError::ObjectNotFound {
+                    obj_type: object_ref.obj_type.to_string(),
+                    obj_id: object_ref.obj_id.clone(),
+                })?
                 .remove(object_ref);
         }
+        Ok(())
     }
 
     /// Apply update operation to the infra cache
-    fn apply_update(&mut self, railjson_obj: &RailjsonObject) {
-        self.apply_delete(&railjson_obj.get_ref());
-        self.apply_create(railjson_obj);
+    fn apply_update(&mut self, railjson_obj: &RailjsonObject) -> Result<()> {
+        self.apply_delete(&railjson_obj.get_ref())?;
+        self.apply_create(railjson_obj)?;
+        Ok(())
     }
 
     /// Apply create operation to the infra cache
-    fn apply_create(&mut self, railjson_obj: &RailjsonObject) {
+    fn apply_create(&mut self, railjson_obj: &RailjsonObject) -> Result<()> {
```

Comment:
I do understand the need for coherence with the other functions... but this function cannot fail at the moment. Does it make sense to make faillible?

Reply:
I can only imagine rejecting dobloons in the future, right now.
You have exposed all the pros/cons I had in mind on that.
Any other opinion is welcome ! :pray:

Reply:
`apply_create` could fails. We could create an objects with an existing ID.

Reply:
Indeed, I didn't notice the `assert!`, so I did a [separate commit](https://github.com/osrd-project/osrd/pull/5844/commits/7477a1bd419c820b4dc2b98116bd29ff4f3a8d61) to remove it and return an error.